qa/rpc-tests/listtransactions.py

```diff
@@ -11,78 +11,151 @@
 
     def setup_nodes(self):
         enable_mocktime()
         return start_nodes(4, self.options.tmpdir)
 
     def run_test(self):
+        self.test_listtransactionsfrom()
+        self.test_listtransactions()
+
+    def test_listtransactionsfrom(self):
+        # Simple send, 0 to 1:
+        self.sync_all()
+        tmp = self.nodes[2].listtransactionsfrom("*", 10000, 0)
+        curpos = len(tmp)
+        txid = self.nodes[2].sendtoaddress(self.nodes[3].getnewaddress(), 0.1)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_blocks()
+
+        # Basic positive test
+        tmp = self.nodes[2].listtransactionsfrom("*", 1, curpos)
+        assert len(tmp) == 1
+        assert tmp[0]["txid"] == txid
+
+        tmp = self.nodes[2].listtransactionsfrom("*", 10, curpos)
+        assert len(tmp) == 1
+
+        # Negative tests
+        # test beyond end of tx list
+        tmp = self.nodes[2].listtransactionsfrom("*", 100, curpos + 100)
+        assert(len(tmp) == 0)
+
+        # test bad input values
+        try:
+            tmp = self.nodes[2].listtransactionsfrom("*", -1, curpos)
+            assert 0
+        except JSONRPCException:
+            pass
+        try:
+            tmp = self.nodes[2].listtransactionsfrom("*", 100, -1)
+            assert 0
+        except JSONRPCException:
+            pass
+
+        # test multiple rows
+        curpos += 1
+
+        txidsA = [self.nodes[2].sendtoaddress(self.nodes[3].getnewaddress(), 0.2), self.nodes[2].sendtoaddress(
+            self.nodes[3].getnewaddress(), 0.3), self.nodes[2].sendtoaddress(self.nodes[3].getnewaddress(), 0.4)]
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_blocks()
+
+        tmp = self.nodes[2].listtransactionsfrom("*", 100, curpos)
+        assert len(tmp) == 3
+        assert tmp[0]["txid"] == txidsA[0]
+        assert tmp[1]["txid"] == txidsA[1]
+        assert tmp[2]["txid"] == txidsA[2]
+
+        txidsB = [self.nodes[2].sendtoaddress(self.nodes[3].getnewaddress(), 0.5), self.nodes[2].sendtoaddress(
+            self.nodes[3].getnewaddress(), 0.6), self.nodes[2].sendtoaddress(self.nodes[3].getnewaddress(), 0.7)]
+
+        tmp = self.nodes[2].listtransactionsfrom("*", 100, curpos)
+        assert len(tmp) == 6
+        assert tmp[0]["txid"] == txidsA[0]
+        assert tmp[1]["txid"] == txidsA[1]
+        assert tmp[2]["txid"] == txidsA[2]
+        assert tmp[3]["txid"] == txidsB[0]
+        assert tmp[4]["txid"] == txidsB[1]
+        assert tmp[5]["txid"] == txidsB[2]
+
+        # test when I advance to the end, I get nothing
+        curpos += len(tmp)
+        tmp = self.nodes[2].listtransactionsfrom("*", 100, curpos)
+        assert tmp == []
+
+    def test_listtransactions(self):
         # Simple send, 0 to 1:
         txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
         self.sync_all()
         assert_array_result(self.nodes[0].listtransactions(),
```

Comment:
I ran "autopep8 --max-line-length 120 --in-place <files>" on this.  We should generally be adhering to a python code formatting standard as well as a c++ one.

Reply:
we should and I think we should introduce an automatic check in travis to verify that the file are properly linted.

what I would do thou is using not mixing formating w/ functional changes, because it makes review more difficult for me. This is not the case of this PR cause is simple and straight forward but still.

Reply:
good point